next permutation in c++

```diff
@@ -0,0 +1,69 @@
+#include <bits/stdc++.h>
+using namespace std;
+
+//to print a vector
+void print(vector<int> a,int n){ 
+	for(int i=0;i<n;i++)
+		cout<<a[i]<<" ";
+	cout<<endl;
+}
+
+void nextPermutation(vector<int> a,int n){
+	int k=-1,l,temp;
```

Comment:
These should be declared on separate lines:
```c++
int k = -1;
int l = 0;
int temp{};  // Alternative, default initializing temp to zero.
```

Note that it is good practice to initialize all you variables with explicit values.